### PR TITLE
Add a new test test_backuptarget_available_during_engine_image_not_ready

### PR DESCRIPTION
**Pull request by jenting**
#### Proposal Change

Add a new test `test_backuptarget_available_during_engine_image_not_ready` to verify configures the backup target setting during the default engine image is not ready.

#### Issues

https://github.com/longhorn/longhorn/issues/2960

### CODE REVIEW

**Comment by jenting** [comment]
Test results https://ci.longhorn.io/job/private/job/longhorn-tests-regression/457/testReport/tests/test_basic/

File: manager/integration/tests/test_basic.py [hunk 1]
```diff
@@ -3701,3 +3702,66 @@ def test_expand_pvc_with_size_round_up(client, core_api, volume_name):  # NOQA
...
+        poll_intervals = ["0", "300"]
+        for poll_interval in poll_intervals:
+            set_backupstore_poll_interval(client, poll_interval)
+
+            # Restart engine image DaemonSet
+            default_img = common.get_default_engine_image(client)
+            ds_name = "engine-image-" + default_img.name
+            apps_api.delete_namespaced_daemon_set(
+                ds_name, common.LONGHORN_NAMESPACE)
+            time.sleep(1)  # Wait for the engine image Pods be deleted
```
Comment (shuo-wu) [inline manager/integration/tests/test_basic.py:3742]: Use a retry loop to validate the default engine image becoming not-ready.

File: manager/integration/tests/test_basic.py [hunk 2]
```diff
@@ -3701,3 +3702,66 @@ def test_expand_pvc_with_size_round_up(client, core_api, volume_name):  # NOQA
...
+
+            # Set a valid backup target during the engine image Pods restart
+            backupsettings = backupstore.split("$")
+            set_backupstore_url(client, url)
+            set_backupstore_credential_secret(client, cred_secret)
+            time.sleep(2)  # Wait for the backup_target_controller
+            bt = common.find_backup_target_default(client)
+            if not bt.available:
+                raise Exception(
+                    'BackupTarget status.available should be True')
```
Comment (shuo-wu) [inline manager/integration/tests/test_basic.py:3752]: Use a retry loop to validate the backup target becoming available. Similar below

File: manager/integration/tests/test_basic.py [hunk 3]
```diff
@@ -3701,3 +3702,66 @@ def test_expand_pvc_with_size_round_up(client, core_api, volume_name):  # NOQA
...
+            default_img = common.get_default_engine_image(client)
+            ds_name = "engine-image-" + default_img.name
+            apps_api.delete_namespaced_daemon_set(
+                ds_name, common.LONGHORN_NAMESPACE)
+            time.sleep(1)  # Wait for the engine image Pods be deleted
+
+            # Set a valid backup target during the engine image Pods restart
+            backupsettings = backupstore.split("$")
+            set_backupstore_url(client, url)
+            set_backupstore_credential_secret(client, cred_secret)
```
Comment (shuo-wu) [inline manager/integration/tests/test_basic.py:3747]: Make sure the backup target is unavailable here.

**Review by shuo-wu — CHANGES_REQUESTED** [review]

**Review by shuo-wu — APPROVED** [review]
Based on the current manager PR, this LGTM. But we need to re-check this if there is any modification introduced in the manager PR.

**Review by innobead — APPROVED** [review]
LGTM

**Review by khushboo-rancher — APPROVED** [review]
LGTM

**Comment by shuo-wu** [comment]
This test is good to go after resolving the conflict.

**Review by shuo-wu — APPROVED** [review]

**Comment by innobead** [comment]
@jenting Please backport to 1.2.x.

**Pull request merged**